delegation/src/Pos/Arbitrary/Delegation.hs

```diff
@@ -19,8 +19,7 @@ import           Pos.Util.Util (leftToPanic)
 
 genDlgPayload :: HasConfiguration => EpochIndex -> Gen DlgPayload
 genDlgPayload epoch =
-    -- TODO [CSL-2173]: Clarify
-    leftToPanic "genDlgPayload: " .
+    leftToPanic "genDlgPayload: generated payload is broken: " .
```

Comment:
How do we know this can't happen?

Reply:
It can. If it happens, we crash.

Reply:
Don't crash with `error` if it's something you expect can happen. Just change the "TODO" from "Clarify" to "Refactor" and I'll take it from there.